scripts/smoke_test.sh

```diff
@@ -0,0 +1,258 @@
+#!/bin/bash
+
+if [[ $1 == "" ]]; then
+  echo "Missing test image!"
+  exit 1
+fi
+
+image_name=$(basename $1)
+echo "=== Smoke testing ${image_name}"
+
+[[ $image_name =~ ^(.+)-(libvirt|ovirt)-([0-9]*)([a-z]*)(.+)\.qc2$ ]]
+product_name=${BASH_REMATCH[1]}              #manageiq
+platform=${BASH_REMATCH[2]}                  #libvirt
+product_version_number=${BASH_REMATCH[3]}    #18? (this is almost always blank, will fill in later)
+product_version_name=${BASH_REMATCH[4]}      #quinteros?
+product_version_remainder=${BASH_REMATCH[5]} #-1-20240403
+# Fill in product version number if missing
+if [[ $product_version_number == "" ]]; then
+  letters=({a..z})
+  letter=${product_version_name::1}
+  for (( i=0; ; i++ ))
+  do
+    if [[ ${letters[$i]} == $letter ]]; then
+      product_version_number=$(($i + 1))
+      break
+    fi
+  done
+fi
+
+script_directory=$(dirname -- "$( readlink -f -- "$0"; )")
+
+vm_name="smoketest-${BASH_REMATCH[3]}${BASH_REMATCH[4]}${BASH_REMATCH[5]}"
+newpassword=smartvm123
+
+echo "=== Copying image..."
+disk_image="/var/lib/libvirt/images/${vm_name}.qc2"
+cp $1 $disk_image
+
+echo "=== Creating database disk"
+db_disk="/var/lib/libvirt/images/${vm_name}-db.qc2"
+qemu-img create -f qcow2 $db_disk 2G
+
+echo "=== Creating VM..."
+virt-install -n $vm_name --memory 4096 --vcpus 2 --cpu host --disk $disk_image --disk $db_disk --network default --graphics spice --osinfo centos-stream9 --import --noautoconsole
+
+echo "=== Waiting for VM to boot..."
+sleep 30
+
+echo "=== Finding IP Address..."
+ip_address=$(virsh guestinfo $vm_name --interface | grep 192.168.122. | cut -f2 -d: | xargs)
+echo "--- Found ${ip_address}"
+
+echo "=== Waiting for SSH..."
+while true; do
+  ncat ${ip_address} 22 < /dev/null && break
+  sleep 1
+done
+sleep 1
+
+echo "=== Adding to known_hosts..."
+cat ~/.ssh/known_hosts |grep -v $ip_address > ~/.ssh/known_hosts
+ssh-keyscan $ip_address >> ~/.ssh/known_hosts
+
+echo "=== Changing default password..."
+/usr/bin/expect <<EOF;
+spawn sshpass -p smartvm ssh -tt root@$ip_address true
+match_max 100000
+expect "New password: "
+send -- "$newpassword\r"
+expect "Retype new password: "
+send -- "$newpassword\r"
+expect eof
+EOF
+
+
+### Configure the appliance
+echo "=== Configuring the appliance..."
+sshpass -p $newpassword ssh -tt root@$ip_address "hostnamectl hostname ${ip_address}.xip.io"
+sshpass -p $newpassword ssh -tt root@$ip_address "echo '${ip_address} ${ip_address}.xip.io 192' >> /etc/hosts"
+/usr/bin/expect <<EOF;
+spawn sshpass -p $newpassword ssh -tt root@$ip_address appliance_console
+match_max 100000
+expect "Press any key to continue.\r"
+send -- " "
+expect -exact "\[H\[2J\[3JAdvanced Setting\r
```

Comment:
No, I wasn't sure if it would be helpful in the future when a change is made to the appliance console and this needs to be edited.  I think I can reduce it and keep the important details.